### ISSUE TITLE
log axis on wrong side

### ISSUE THREAD

**Issue by lindsayplatt** [USGS-R/gsplot#414]
gsplot object is correct, but plotting the wrong thing

- Might have something to do with copy if not explicit (JW)
- Maybe because print is weird???

```
# this should only have side 4 logged, the helper functions say the right thing, but the plot shows both logged
gs <- gsplot() %>% 
    points(1:2,c(2,25)) %>% 
    points(c(1,2), c(1, 1000000000), side=4, log='y') 
gs

logged(gs, side=2)
[1] FALSE
logged(gs, side=4)
[1] TRUE
```

![image](https://cloud.githubusercontent.com/assets/13220910/20079274/890b3810-a509-11e6-8d63-5005561c1daa.png)

**Comment by lindsayplatt** [comment]
In print, in the side for loop we don't want to set `old.par <- par(x[[side.name]]$par)` . This should be done in two steps, or par() should return the old par before it was modified.

**Comment by lindsayplatt** [comment]
I would like to be able to log each side separately, but that pops up with warnings currently. This should log the right axis, but not the left. I think it might be related to this issue.

``` r
gs <- gsplot() %>% 
    axis(side=4) %>% 
    view(side=2, ylim=c(1,100)) %>% 
    view(side=4, ylim=c(1,100), log='y') 

Warning messages:
1: In doTryCatch(return(expr), name, parentenv, handler) :
  nonfinite axis limits [GScale(-inf,0,2, .); log=1]
2: In doTryCatch(return(expr), name, parentenv, handler) :
  nonfinite axis limits [GScale(-inf,0,2, .); log=1]

gs

Warning message:
In plot.window(xlim = lim(object, side), ylim = lim(object, side),  :
  nonfinite axis limits [GScale(-inf,0,2, .); log=1]
```

**Comment by jordansread** [comment]
does it work if you have data on those sides?

**Comment by lindsayplatt** [comment]
Oh I guess it eliminates the warnings, but it logs both axis instead of just one. It's also not assigning the correct side as logged.

``` r
gs <- gsplot() %>% 
    points(1:2, c(1,100), side=2) %>% 
    points(3:4, c(10,1000), side=4) %>% 
    axis(side=4) %>% 
    view(side=2, ylim=c(1,100)) %>% 
    view(side=4, ylim=c(1,100), log='y') 
gs

> logged(gs, side=1)
[1] FALSE
> logged(gs, side=2)
[1] TRUE
> logged(gs, side=4)
[1] FALSE
```

![image](https://cloud.githubusercontent.com/assets/13220910/20309207/b761b9ba-ab0c-11e6-91c5-c8a857da52ef.png)

**Comment by lindsayplatt** [comment]
Reset par because that was causing an issue w/ always showing a logged axis when only one should be. Then we thought it had something to do with just view, but there are still logging issues when you don't use view.

### Log side 4, but not side 2

[this works]

``` r
gs <- gsplot(xaxs='r', yaxs='r') %>% 
    points(1:3, c(1,10,100), side=2) %>% 
    points(3:5, c(10,100,1000), col="blue", side=4, log='y') %>% 
    axis(side=4)
gs

> logged(gs)
$side.1
[1] FALSE

$side.2
[1] FALSE

$side.4
[1] TRUE
```

![image](https://cloud.githubusercontent.com/assets/13220910/20365370/49f0dc5c-ac0c-11e6-8036-bfa43dd92570.png)

### Log side 2, but not side 4

[correct object, correct plot data, incorrect side 4 axis]

``` r
gs <- gsplot(xaxs='r', yaxs='r') %>% 
    points(1:3, c(1,10,100), side=2, log='y') %>% 
    points(3:5, c(10,100,1000), col="blue", side=4) %>% 
    axis(side=4)
gs

> logged(gs)
$side.1
[1] FALSE

$side.2
[1] TRUE

$side.4
[1] FALSE
```

![image](https://cloud.githubusercontent.com/assets/13220910/20365420/6bce2afa-ac0c-11e6-960d-b363da17dd0f.png)

**Comment by lindsayplatt** [comment]
[This commit](https://github.com/USGS-R/gsplot/pull/418/commits/3a9c852a0a8d4c670f3e24eedec1859b8bb56cd5) resets so that side 4 is not logged, but you lose the labels.

``` r
gs <- gsplot(xaxs='r', yaxs='r') %>% 
    points(1:3, c(1,10,100), side=2, log='y') %>% 
    points(3:5, c(10,100,1000), col="blue", side=4) %>% 
    axis(side=4)
gs
```

![image](https://cloud.githubusercontent.com/assets/13220910/20367303/8ee978b6-ac14-11e6-9561-a6957e2ccfae.png)

**Comment by lindsayplatt** [comment]
It's not something wrong with the y axes specifically, but with the second and third axes drawn. I changed the order the sides were drawn (2,4,1 instead), and it's the second and third sides drawn that are missing labels. So, this must mean it's something to do with our setting/re-setting par. 

![image](https://cloud.githubusercontent.com/assets/13220910/20541907/b9a22602-b0c4-11e6-9a12-fc32074e9bf4.png)

**Comment by jordansread** [comment]
or the `tmp` code in `draw_axis`? Have you looked at that?

**Comment by lindsayplatt** [comment]
Gah - keep forgetting about that tmp stuff. Going to search now

**Comment by lindsayplatt** [comment]
`tmp` isn't even used in the example above because there's only one call to each axis. I've compared par() right before the `do.call('Axis', axis.args)` line for when side 1 is called first and then when it's called second. No difference.

**Comment by jordansread** [comment]
so you checked to verify that the logic is correct for that `tmp` code and that it isn't evaluated anyhow?

**Comment by lindsayplatt** [comment]
I knew it didn't get evaluated because I put a breakpoint in there and it never stopped.

**Comment by jordansread** [comment]
OK, just want to rule it out completely - even if you put `message('tmp used')` in there, it doesn't hit that? Sometimes breakpoints are goofy in this package. Alternatively, `debug(draw_axis)`?

**Comment by lindsayplatt** [comment]
Correct, just tried that.

**Comment by lindsayplatt** [comment]
I reverted back trying out different commits (as far as #400) w/ the changes I made in [this commit](https://github.com/USGS-R/gsplot/pull/418/commits/3a9c852a0a8d4c670f3e24eedec1859b8bb56cd5). Once I made the changes, the labels disappear so that issue is something that I introduced....now what about where I put `par` calls is causing that :thinking:

**Comment by lindsayplatt** [comment]
.......................still stumped.

**Comment by lindsayplatt** [comment]
First time I call the plot, neither y-axis shows up. Second time (as long as I don't clear it), they both show up, but the axis ticks & labels for side=4 are still wrong. Seems like there is something going on with all of the `par(new=TRUE)`

**Comment by lindsayplatt** [comment]
@jread-usgs to summarize what's going on here:

1. When trying to log one axis independent of the axis of the same direction (1+3, or 2+4), both are logged. This is likely because of some par refresh that is not happening, so `log='x'` and `log='y'` propagates through to the next axis call. Using `view` to specify what to log has a similar effect. Setting `log=''` does not do anything either. Even though `set_side_log` looks like it should handle that appropriately. 

These graphs are trying to have `side=3` logged, but `side=1` not logged.
```r
# These two graphs label both side 1 and 3 as logged
# BUT they only log the side 3 data
gs <- gsplot() %>% 
  points(c(1,10,100), 1:3, side=1) %>% 
  points(c(10,100,1000), 3:5, col="blue", side=3, log='x') %>% 
  axis(side=3)
gs

gs <- gsplot() %>% 
  points(c(1,10,100), 1:3, side=1, log='') %>% 
  points(c(10,100,1000), 3:5, col="blue", side=3, log='x') %>% 
  axis(side=3)
gs

# both graphs say the correct logic behind-the-scenes
logged(gs, side=1)
[1] FALSE
logged(gs, side=3)
[1] TRUE
```
![image](https://cloud.githubusercontent.com/assets/13220910/22518934/29156f9e-e874-11e6-9245-6e52e2f0b076.png)
```r
# These two graphs label both side 1 and 3 as logged
# AND log both side 1 and 3 data
gs <- gsplot() %>% 
  points(c(1,10,100), 1:3, side=1) %>% 
  points(c(10,100,1000), 3:5, col="blue", side=3) %>% 
  axis(side=3) %>% 
  view(side=3, log='x')
gs

gs <- gsplot() %>% 
  points(c(1,10,100), 1:3, side=1) %>% 
  points(c(10,100,1000), 3:5, col="blue", side=3) %>% 
  axis(side=3) %>% 
  view(side=1, log='') %>%
  view(side=3, log='x')
gs

# these graphs both say the exact opposite of what should be true:
logged(gs, side=1)
[1] TRUE
logged(gs, side=3)
[1] FALSE
```
![image](https://cloud.githubusercontent.com/assets/13220910/22518984/5371ba2c-e874-11e6-9784-3f4a30150028.png)

2. I implemented a change to resetting pars in [this commit](https://github.com/USGS-R/gsplot/pull/418/commits/3a9c852a0a8d4c670f3e24eedec1859b8bb56cd5). This resulted in axes correctly being logged! BUT the labels were lost.

```r
# Correctly logged data, but no labels for any side but 1
# AND side 1 labels are logged
gs <- gsplot() %>% 
    points(c(1,10,100), 1:3, side=1) %>% 
    points(c(10,100,1000), 3:5, col="blue", side=3, log='x') %>% 
    axis(side=3) %>% 
    axis(side=4)
gs

# checks out correctly again
logged(gs, side=1)
[1] FALSE
logged(gs, side=3)
[1] TRUE
```
![image](https://cloud.githubusercontent.com/assets/13220910/22519403/bea7918a-e875-11e6-9f60-45ead2f77fdf.png)

3. I just added [this commit](https://github.com/lindsaycarr/gsplot/commit/f9450cc9d6501573fb705407e4b6dd773e19a785), which makes the plot where you log within the points call correct, but the logging using view is doing the exact opposite (but there are labels!).

```r
gs <- gsplot() %>% 
  points(c(1,10,100), 1:3, side=1) %>% 
  points(c(10,100,1000), 3:5, col="blue", side=3, log='x') %>% 
  axis(side=3)
gs

gs <- gsplot() %>% 
  points(c(1,10,100), 1:3, side=1, log='') %>% 
  points(c(10,100,1000), 3:5, col="blue", side=3, log='x') %>% 
  axis(side=3)
gs

logged(gs, side=1)
[1] FALSE
logged(gs, side=3)
[1] TRUE
```
![image](https://cloud.githubusercontent.com/assets/13220910/22519574/659cc8ac-e876-11e6-831b-c73446e017e7.png)

```r
# side 1 data are logged, but side 3 data don't
# both axes are labeled as logged
gs <- gsplot() %>% 
    points(c(1,10,100), 1:3, side=1) %>% 
    points(c(10,100,1000), 3:5, col="blue", side=3) %>% 
    axis(side=3) %>% 
    view(side=3, log='x')
gs

# and that appears to be the behavior it expects
logged(gs, side=1)
[1] TRUE
logged(gs, side=3)
[1] FALSE
```
![image](https://cloud.githubusercontent.com/assets/13220910/22519746/273156f4-e877-11e6-970d-f662823f2efa.png)

So in summary - with the changes to `print.R` that are currently on my `axisLogWrongSide` branch, it all seems to be working except for trying to log w/ `view()`.

**Comment by lindsayplatt** [comment]
Ok, I believe I figured this out. The `view.gsplot` function takes `side` as an argument (with default `c(1,2)`) BUT was passing the arg `side=c(1,2)` into `gather_function_info`. So if `side=3`, it would always get passed in as `c(1,2)`. See [this commit](https://github.com/lindsaycarr/gsplot/commit/3c3ad881b886bf98232cc05c087b9ca76de0e739) for the change.

**Comment by jordansread** [comment]
Good find @lindsaycarr !

**Comment by lindsayplatt** [comment]
Fixed with #424 !!!!

It was those pesky pars - needed to reset only the pars that pertained to the side, but not all pars.
